data-generation/src/main/scala/thesis/SnapshotDelta.scala

```diff
@@ -68,8 +68,29 @@ class SnapshotDelta(val graphs: MutableList[Snapshot],
       forwardApplyLogs(closestGraph.graph, logsToApply)
     }
   }
-  
-  override def directNeighbours(vertexId: VertexId, interval: Interval): RDD[VertexId] = throw new NotImplementedError()
+
+  override def directNeighbours(vertexId: VertexId, interval: Interval): RDD[VertexId] = {
+    val filteredLogs = logs.filter(log => log.action == CREATE || log.action == DELETE)
+      .filter(log => log.entity match {
+        case VERTEX(_) => false
```

Comment:
ikke så viktig, men jeg liker denne syntaksen for å vise at vi kun matcher på typen 
```suggestion
        case _:VERTEX => false
```